integrationTests/testInitializer.go

```diff
@@ -622,3 +647,497 @@ func extractUint64ValueFromTxHandler(txHandler data.TransactionHandler) uint64 {
+func ComputeAndRequestMissingTransactions(
+	nodes []*TestProcessorNode,
+	generatedTxHashes [][]byte,
+	shardResolver uint32,
+	shardRequesters ...uint32,
+) {
+	for _, n := range nodes {
+		if !Uint32InSlice(n.ShardCoordinator.SelfId(), shardRequesters) {
+			continue
+		}
+
+		neededTxs := getMissingTxsForNode(n, generatedTxHashes)
+		requestMissingTransactions(n, shardResolver, neededTxs)
+	}
+}
+
+func getMissingTxsForNode(n *TestProcessorNode, generatedTxHashes [][]byte) [][]byte {
+	neededTxs := make([][]byte, 0)
+
+	for i := 0; i < len(generatedTxHashes); i++ {
+		_, ok := n.ShardDataPool.Transactions().SearchFirstData(generatedTxHashes[i])
+		if !ok {
+			//tx is still missing
+			neededTxs = append(neededTxs, generatedTxHashes[i])
+		}
+	}
+
+	return neededTxs
+}
+
+func requestMissingTransactions(n *TestProcessorNode, shardResolver uint32, neededTxs [][]byte) {
+	txResolver, _ := n.ResolverFinder.CrossShardResolver(procFactory.TransactionTopic, shardResolver)
+
+	for i := 0; i < len(neededTxs); i++ {
+		_ = txResolver.RequestDataFromHash(neededTxs[i])
+	}
+}
+
+// CreateRequesterDataPool creates a datapool with a mock txPool
+func CreateRequesterDataPool(
+	t *testing.T,
+	recvTxs map[int]map[string]struct{},
+	mutRecvTxs *sync.Mutex,
+	nodeIndex int,
+) dataRetriever.PoolsHolder {
+
+	//not allowed to request data from the same shard
+	return CreateTestShardDataPool(
+		&mock.ShardedDataStub{
+			SearchFirstDataCalled: func(key []byte) (value interface{}, ok bool) {
+				assert.Fail(t, "same-shard requesters should not be queried")
+				return nil, false
+			},
+			ShardDataStoreCalled: func(cacheId string) (c storage.Cacher) {
+				assert.Fail(t, "same-shard requesters should not be queried")
+				return nil
+			},
+			AddDataCalled: func(key []byte, data interface{}, cacheId string) {
+				mutRecvTxs.Lock()
+				defer mutRecvTxs.Unlock()
+
+				txMap := recvTxs[nodeIndex]
+				if txMap == nil {
+					txMap = make(map[string]struct{})
+					recvTxs[nodeIndex] = txMap
+				}
+
+				txMap[string(key)] = struct{}{}
+			},
+			RegisterHandlerCalled: func(i func(key []byte)) {
+			},
+		},
+	)
+}
+
+// CreateResolversDataPool creates a datapool containing a given number of transactions
+func CreateResolversDataPool(
+	t *testing.T,
+	maxTxs int,
+	senderShardID uint32,
+	recvShardId uint32,
+	shardCoordinator sharding.Coordinator,
+) (dataRetriever.PoolsHolder, [][]byte) {
+
+	txHashes := make([][]byte, maxTxs)
+
+	txPool, _ := shardedData.NewShardedData(storageUnit.CacheConfig{Size: 100, Type: storageUnit.LRUCache})
+
+	for i := 0; i < maxTxs; i++ {
+		tx, txHash := generateValidTx(t, shardCoordinator, senderShardID, recvShardId)
+		cacherIdentifier := process.ShardCacherIdentifier(1, 0)
+		txPool.AddData(txHash, tx, cacherIdentifier)
+		txHashes[i] = txHash
+	}
+
+	return CreateTestShardDataPool(txPool), txHashes
+}
+
+func generateValidTx(
```

Comment:
instead of this func, might add a new func called GenerateTx(receiver, value) in testWalletAccount?

Reply:
Can't do that, because this func is used for creating an already populated datapool, and passing it to the node.